source/material_model/visco_plastic.cc

```diff
@@ -309,13 +309,56 @@ namespace aspect
 
       if (this->get_parameters().enable_elasticity)
         {
+          // Fill the force outputs with the body force term for the RHS.
           rheology->elastic_rheology.fill_elastic_force_outputs(in, average_elastic_shear_moduli, out);
+          // Fill the reaction terms that account for the rotation of the stresses.
           rheology->elastic_rheology.fill_reaction_outputs(in, average_elastic_shear_moduli, out);
+          // Fill the reaction_rates that apply the stress update of the previous
+          // timestep to the advected and rotated stress computed in the previous timestep ($\tau^{0}$)
+          // to obtain $\tau^{t}$.
+          // Only fill them if operator splitting is used for fields or when the particles track
+          // the visco-elastic stresses.
+          if (this->get_parameters().use_operator_splitting ||
+              ((this->get_parameters().mapped_particle_properties).count(this->introspection().compositional_index_for_name("ve_stress_xx"))))
+            rheology->elastic_rheology.fill_reaction_rates(in, average_elastic_shear_moduli, out);
         }
     }
 
 
 
+    template <int dim>
+    double
+    ViscoPlastic<dim>::
+    get_elastic_viscosity(const double shear_modulus) const
+    {
+      if (this->get_parameters().enable_elasticity)
+        {
+          //TODO Check: Should this viscosity be scaled with the timestep ratio?
```

Comment:
I guess the question is whether to modify the value here, or on each of the lines calling this function. I think you already made this decision in `material_model/rheology/visco_plastic.cc`?

Need to address this in `viscoelastic.cc` too

Reply:
Yes, that's the question. calculate_elastic_viscosity in material_model/rheology/elasticity.cc does not scale with the timestep ratio, so in the rest of elasticity.cc it is done after calling calculate_elastic_viscosity. The function here, get_elastic_viscosity, is called by visualization postprocessors. It is less error-prone to do it in one place, which would be in calculate_elastic_viscosity in elasticity.cc. Would you agree?

Reply:
Yes, I agree. I think that everywhere in the derivation the "elastic viscosity" term actually uses the computational timestep (i.e. Eq.51 in our draft), and that has the same right to being called the elastic viscosity as anything else.

Reply:
Okay, done! I introduced a new function for the timestep ratio and scaled the elastic viscosity in the calculate_elastic_viscosity function, this cleaned up some duplicated code in several places.